src/nodes.coffee

```diff
@@ -2095,13 +2095,15 @@ exports.For = class For extends While
     @body    = Block.wrap [body]
     @own     = !!source.own
     @object  = !!source.object
-    [@name, @index] = [@index, @name] if @object
+    @from    = !!source.from
+    [@name, @index] = [@index, @name] if @object or @from
     @index.error 'index cannot be a pattern matching expression' if @index instanceof Value
     @range   = @source instanceof Value and @source.base instanceof Range and not @source.properties.length
     @pattern = @name instanceof Value
     @index.error 'indexes do not apply to range loops' if @range and @index
```

Comment:
this can be an issue:

```
./bin/coffee -bce '1 for a from [1..5]'
[stdin]:1:7: error: indexes do not apply to range loops
1 for a from [1..5]
```